Lib/test/test_dataclasses.py

```diff
@@ -1934,6 +1934,18 @@ class R:
                     self.assertEqual(new_sample.x, another_new_sample.x)
                     self.assertEqual(sample.y, another_new_sample.y)
 
+    def test_dataclasses_qualnames(self):
+        @dataclass
+        class A:
+            x: int
+            y: int
+
+        self.assertEqual(A.__init__.__name__, "__init__")
+        self.assertEqual(A.__init__.__qualname__, "TestCase.test_dataclasses_qualnames.<locals>.A.__init__")
```

Comment:
I would add tests for all generated methods: `__init__`, `__repr__`, `__eq__`, `__lt__`, etc.

Reply:
Thanks for noticing that, looks like `__hash__` function doesn't pass through the `_set_attribute` like others so had to fix that too!